app/ts/TradingUI/Trading.tsx

```diff
@@ -57,16 +58,59 @@ const TradingView = ({ maybeReadClient, maybeWriteClient, marketData, currentTim
 		}
 	})
 
-	const refresh = async () => {
+	const updateSharesOut = async (daiInput: bigint) => {
 		if (maybeWriteClient.deepValue === undefined) return
 		if (maybeReadClient.deepValue === undefined) return
 		if (marketData.deepValue === undefined) return
+		const baseSharesExpected = daiInput / marketData.deepValue.numTicks
+		const expectedSwapShares = await expectedSharesAfterSwap(maybeReadClient.deepValue, marketData.deepValue.marketAddress, yesSelected.value === 'No', baseSharesExpected)
+		expectedSharesOut.deepValue = baseSharesExpected + expectedSwapShares
+	}
+
+	useSignalEffect(() => {
+		if (maybeWriteClient.deepValue === undefined) return
+		if (maybeReadClient.deepValue === undefined) return
+		if (marketData.deepValue === undefined) return
+
+		if (buySelected.value === 'Buy') {
+			if (daiInputAmountToBuy.deepValue === undefined) return
+			// TODO, fix race conditions
+			updateSharesOut(daiInputAmountToBuy.deepValue)
+		} else {
+			if (yesSelected.value === 'Yes') {
+				console.error('TODO: not implemented')
+			} else {
+				console.error('TODO: not implemented')
```

Comment:
Any reason to not implement these now?

Reply:
I don't have those functions in the contract

Reply:
Oh, OK.  That is fine then.